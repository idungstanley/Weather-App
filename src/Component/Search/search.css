.search {
  width: 100%;
  height: 30px;
  height: 40px;
  border: none;
}

.search:focus {
  outline: none;
  border: none;
  padding: 5px;
}

form {
  margin-top: 3px;
  width: 100%;
  align-items: center;
}
