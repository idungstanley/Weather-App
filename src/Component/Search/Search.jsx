import React from 'react';
import PropTypes from 'prop-types';
import './search.css';
import PerCountry from '../Card/PerCountry';

const Search = ({ getvalue, onChange }) => (
  <form>
    <input
      className="search"
      type="search"
      placeholder="Search..."
      name=""
      id=""
      value={getvalue}
      onChange={onChange}
    />
    <PerCountry />
  </form>
);

Search.propTypes = {
  getvalue: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
export default Search;
