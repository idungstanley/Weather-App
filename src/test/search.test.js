import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../Redux/store";
import Search from "../Component/Search/Search";

describe("Render HomePage component", () => {
  test("render HomePage", () => {
    const tree = renderer
      .create(
        <Router>
          <Provider store={store}>
            <Search />
          </Provider>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
