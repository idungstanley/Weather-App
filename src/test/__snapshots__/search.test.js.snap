// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render HomePage component render HomePage 1`] = `
<form>
  <input
    className="search"
    id=""
    name=""
    placeholder="Search..."
    type="search"
  />
</form>
`;
